src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2606,9 +2600,21 @@ private CorInfoHelpFunc getSharedCCtorHelper(CORINFO_CLASS_STRUCT_* clsHnd)
 
         private CORINFO_CLASS_STRUCT_* getTypeForBoxOnStack(CORINFO_CLASS_STRUCT_* cls)
         {
-            // Todo: implement...
-            _ = HandleToObject(cls);
-            return null;
+            TypeDesc clsTypeDesc = HandleToObject(cls);
+            if (clsTypeDesc.IsNullable)
+            {
+                clsTypeDesc = clsTypeDesc.Instantiation[0];
+            }
+
+            if (clsTypeDesc.RequiresAlign8())
+            {
+                // Conservatively give up on such types (32bit)
+                return null;
+            }
+
+            // Instantiate StackAllocatedBox<T> helper type with the type we're boxing
+            MetadataType placeholderType = _compilation.TypeSystemContext.SystemModule.GetKnownType("System.Runtime.CompilerServices", "StackAllocatedBox`1");
```

Comment:
```suggestion
            MetadataType placeholderType = _compilation.TypeSystemContext.SystemModule.GetType("System.Runtime.CompilerServices", "StackAllocatedBox`1", throwIfNotFound: false);
            if (placeholderType == null)
            {
                // Give up if corelib does not have support for stackallocation
                return null;
            }
```
Should we handle the experimental mini-corelibs gracefully? It is double-edged sword... .